src/NLog/LayoutRenderers/NdlcTimingLayoutRenderer.cs

```diff
@@ -0,0 +1,97 @@
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.LayoutRenderers
+{
+#if !SILVERLIGHT
+    using System;
+    using System.Text;
+
+    /// <summary>
+    /// <see cref="NestedDiagnosticsLogicalContext"/> Timing Renderer (Async scope)
+    /// </summary>
+    [LayoutRenderer("ndlctiming")]
+    public class NdlcTimingLayoutRenderer : LayoutRenderer
+    {
+        /// <summary>
+        /// Gets or sets whether to only include the duration of the last scope created
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+        public bool CurrentScope { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether to just display the scope creation time, and not the duration
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+        public bool ScopeBeginTime { get; set; }
+
+        /// <summary>
+        /// Gets or sets the TimeSpan format. Can be any argument accepted by TimeSpan.ToString(format).
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+        public string Format { get; set; }
+
+        /// <summary>
+        /// Renders the timing details of the Nested Logical Context item and appends it to the specified <see cref="StringBuilder" />.
+        /// </summary>
+        /// <param name="builder">The <see cref="StringBuilder"/> to append the rendered data to.</param>
+        /// <param name="logEvent">Logging event.</param>
+        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
+        {
+            DateTime scopeBegin = CurrentScope ? NestedDiagnosticsLogicalContext.PeekCurrentScopeBeginTime() : NestedDiagnosticsLogicalContext.PeekTotalScopeBeginTime();
+            if (scopeBegin != DateTime.MinValue)
+            {
+                if (ScopeBeginTime)
+                {
+                    var formatProvider = GetFormatProvider(logEvent, null);
+                    scopeBegin = Time.TimeSource.Current.FromSystemTime(scopeBegin);
+                    builder.Append(scopeBegin.ToString(Format, formatProvider));
+                }
+                else
+                {
+                    TimeSpan duration = scopeBegin != DateTime.MinValue ? DateTime.UtcNow - scopeBegin : TimeSpan.Zero;
+                    if (duration < TimeSpan.Zero)
+                        duration = TimeSpan.Zero;
+
+                    var formatProvider = GetFormatProvider(logEvent, null);
+#if !NET3_5
+                    builder.Append(duration.ToString(Format, formatProvider));
+#else
+                    builder.Append(duration.ToString());
```

Comment:
formatProvider not needed here?

Reply:
There is no TimeSpan.ToString with Format or formatProvider in NET3_5

Reply:
Indeed, but I think the call ` var formatProvider = GetFormatProvider(logEvent, null);` is not needed.

Reply:
Fixed